packages/sdk/android/src/main/java/com/primerioreactnative/components/manager/redirect/PrimerRNHeadlessUniversalCheckoutComponentWithRedirectManager.kt

```diff
@@ -0,0 +1,250 @@
+import io.primer.android.components.manager.banks.composable.BanksStep
+import io.primer.android.components.manager.componentWithRedirect.PrimerHeadlessUniversalCheckoutComponentWithRedirectManager
+import io.primer.android.components.manager.componentWithRedirect.component.BanksComponent
+import io.primer.android.components.manager.core.composable.PrimerValidationStatus
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import org.json.JSONArray
+import org.json.JSONObject
+import kotlinx.coroutines.coroutineScope
+
+class PrimerRNHeadlessUniversalCheckoutComponentWithRedirectManager(
+  private val reactContext: ReactApplicationContext,
+) : ReactContextBaseJavaModule(reactContext) {
+  override fun getName() = "RNTPrimerHeadlessUniversalCheckoutComponentWithRedirectManager"
+
+  private var job: Job? = null
+  private var banksComponent: BanksComponent? = null
+
+  @ReactMethod
+  fun configure(paymentMethodType: String, promise: Promise) {
+
+    val viewModelStoreOwner = reactContext.currentActivity as? ViewModelStoreOwner ?: run {
+      PrimerRNViewModelStoreOwner()
+    }
+
+    banksComponent =
+      PrimerHeadlessUniversalCheckoutComponentWithRedirectManager(viewModelStoreOwner).provide<BanksComponent>(
+        paymentMethodType = paymentMethodType
+      )
+
+    val lifecycleOwner = reactContext.currentActivity as? LifecycleOwner ?: run {
+      promise.reject("LifecycleOwner", "Expected activity class that implement LifecycleOwner")
+      return
+    }
+
+    job = lifecycleOwner.lifecycleScope.launch {
+      banksComponent?.start()
+      coroutineScope {
+        launch {
+          configureBanksListener()
+        }
+
+        launch {
+          configureValidationListener()
+        }
+
+        launch {
+          configureErrorListener()
+        }
+        promise.resolve(null)
+      }
+
+    }
+
+  }
+
+  @ReactMethod
+  fun onBankSelected(bankId: String, promise: Promise) {
+    banksComponent?.updateCollectedData(BanksCollectableData.BankId(bankId))
+    promise.resolve(null)
+
+  }
+
+  @ReactMethod
+  fun onBankFilterChange(filter: String, promise: Promise) {
+    banksComponent?.updateCollectedData(BanksCollectableData.Filter(filter))
+    promise.resolve(null)
+  }
+
+  private suspend fun configureErrorListener() {
+    banksComponent?.componentError?.collectLatest { error ->
+
+      sendEvent(
+        PrimerHeadlessUniversalCheckoutRedirectManagerEvent.ON_ERROR.eventName,
+        JSONObject().apply {
+          put(
+            "errors", JSONArray(
+              Json.encodeToString(error)
+            )
+          )
+        }
+      )
+    }
+  }
+
+  private suspend fun configureBanksListener() {
+    banksComponent?.componentStep?.collectLatest { banksStep ->
```

Comment:
[q]: should we reject the promise in case `banksComponent` is null?

Reply:
This is not a promise function. So, we don't need to reject here. It'll automatically handle if null in onBankFilterChange or onBankSelected.